azext_iot/tests/test_iot_pnp_int.py

```diff
@@ -0,0 +1,183 @@
+# coding=utf-8
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Unpublished works.
+# --------------------------------------------------------------------------------------------
+import pytest
+import random
+import json
+import os
+import sys
+
+from os.path import exists
+from uuid import uuid4
+from azure.cli.testsdk import LiveScenarioTest
+from azure.cli.core.util import read_file_content
+
+
+# Add test tools to path
+sys.path.append(os.path.abspath(os.path.join('.', 'iotext_test_tools')))
+
+# Set these to the proper PnP Endpoint, PnP Cstring and PnP Repository for Live Integration Tests.
+_endpoint = os.environ.get('azext_pnp_endpoint')
+_repo_id = os.environ.get('azext_pnp_repository')
+_repo_cs = os.environ.get('azext_pnp_cs')
+
+_interface_payload = 'test_pnp_create_payload_interface.json'
+_capability_model_payload = 'test_pnp_create_payload_model.json'
+
+if not all([_endpoint, _repo_id, _repo_cs]):
+    raise ValueError('Set azext_pnp_endpoint, azext_pnp_repository and azext_pnp_cs to run PnP model integration tests.')
+
+
+def change_dir():
+    from inspect import getsourcefile
+    os.chdir(os.path.dirname(os.path.abspath(getsourcefile(lambda: 0))))
+
+
+class TestPnPModel(LiveScenarioTest):
+
+    LiveScenarioTest.rand_val = random.randint(1, 10001)
```

Comment:
You don't need to set this on the base class. You should be able to have just `rand_val = random.randint(1, 10001)` and be able to reference `TestPnPModel.rand_val`.

Reply:
Done